src/mod_ir_temp.rs

```diff
@@ -0,0 +1,263 @@
+/// This module implements the functions of the [MOD-IR-TEMP](https://www.olimex.com/Products/Modules/Sensors/MOD-IR-TEMP/open-source-hardware)
+use i2cdev::linux::LinuxI2CError;
+use linux_embedded_hal::{Delay, I2cdev};
+use mlx9061x::ic::Mlx90614;
+use mlx9061x::{Error, Mlx9061x, SlaveAddr};
+
+const THRESHOLD_AMBIENT: f32 = 22.0;
+const THRESHOLD_OBJECT: f32 = 44.0;
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum IrTempStateChange {
+	None,
+	ChangedToAmbientToHot,
+	ChangedToObjectToHot,
+	ChanedToBothToHot,
+	ChangedToCancelled,
+}
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum IrTempState {
+	Normal,
+	TooHot,
+}
+
+pub struct ModIrTemps {
```

Comment:
Please add some documentation to the individual elements.

```suggestion
pub struct ModIR {
```